diff --git a/guix/ui.scm b/guix/ui.scm
index d6d5eb9dcd..2a29aae9a4 100644
--- a/guix/ui.scm
+++ b/guix/ui.scm
@@ -368,8 +368,8 @@ (define* (report-unbound-variable-error args #:key frame)
        (#f
         (display-hint (G_ "Did you forget a @code{use-modules} form?")))
        ((? module? module)
-        (display-hint (G_ "Did you forget @code{(use-modules ~a)}?")
-                      (module-name module)))))))
+        (display-hint (G_ "Did you forget @code{(use-modules ~a)} or @code{#:use-module ~a}?")
+                      (module-name module) (module-name module)))))))
 
 (define (check-module-matches-file module file)
   "Check whether FILE starts with 'define-module MODULE' and print a hint if
